programs/benches/dict_snapshot.c

```diff
@@ -0,0 +1,38 @@
+#include <assert.h>
+#include <stdint.h>
+
+
+typedef struct dict_snapshot_return_values
+{
+    uint64_t range_check_counter;
+    uint64_t segment_arena_counter;
+    uint64_t remaining_gas;
+    struct {
+        uint8_t discriminant;
+        struct {
+            void *ptr;
+            uint32_t start;
+            uint32_t end;
+            uint32_t cap;
+        } err;
+    } result;
+} dict_snapshot_return_values_t;
+
+static void run_bench(dict_snapshot_return_values_t *, uint64_t, uint64_t, uint64_t)
+    __attribute__((weakref("_mlir_ciface_dict_snapshot::dict_snapshot::main(f4)")));
```

Comment:
Just curious, why is it "f4"? Or is it just a name?

Reply:
The name is `_mlir_ciface_dict_snapshot::dict_snapshot::main`, but the `(f4)` exists to differentiate between the main function and the autogenerated loops (for those that have them). Otherwise there'd be name collisions. The number is the same as the one used for contracts where we do not know the function names.

Reply:
Oh I see, thanks!